client/src/components/dom/Auth/Auth.tsx

```diff
@@ -1,34 +1,38 @@
-import * as React from "react";
-import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useState } from "react";
 import styles from "./Auth.module.scss";
-import { SignInWithPasswordCredentials } from "@supabase/supabase-js";
+import AuthModal from "./AuthModal";
+import { useSessionContext } from "@supabase/auth-helpers-react";
 
-function Auth() {
-  const supabase = useSupabaseClient();
-
-  const handleRegister = React.useCallback(() => {
-    if (!supabase) return;
+export enum AuthAction {
+  Login = "Login",
+  Logout = "Logout",
+  Register = "Register",
+  Idle = "Idle",
```

Comment:
Semantically, 'idle' seems a bit misleading to me. I would suggest using something like `hidden` or `inactive` or similar.